packages/template-retail-react-app/app/pages/product-detail/index.test.js

```diff
@@ -63,4 +264,68 @@ test('should render product details page', async () => {
     expect(screen.getAllByText(/14.99/).length).toEqual(2)
     expect(screen.getAllByText(/Add to Cart/).length).toEqual(2)
     expect(screen.getAllByText(/Add to Wishlist/).length).toEqual(2)
+    expect(screen.getAllByTestId('product-view').length).toEqual(1)
+})
+
+describe('product set', () => {
+    test('render multi-product layout', async () => {
+        renderWithProviders(<MockedPageWithProductSet />)
+
+        await waitFor(() => {
```

Comment:
Technically, I don't need to wait here. But I'm foreseeing the time when the hooks-integration work is merged. By then, the page-level component would use hooks to fetch data. And this asynchronous behaviour would require calling `waitFor`.

Reply:
I'm thinking most things should `await` / `waitFor` for that reason. Not many of our components are "dumb" enough to not need the pattern